drivers/video/Kconfig.mcux_mipi_csi2rx

```diff
@@ -0,0 +1,10 @@
+# NXP MIPI CSI-2 Rx driver configuration option
+
+# Copyright NXP 2024
+# SPDX-License-Identifier: Apache-2.0
+
+config VIDEO_MCUX_MIPI_CSI2RX
+	bool "NXP MIPI CSI-2 Rx driver"
+	default y
+	depends on DT_HAS_NXP_MIPI_CSI2RX_ENABLED
+	select VIDEO_MCUX_CSI
```

Comment:
Yes, but it is done in https://github.com/zephyrproject-rtos/zephyr/pull/71860

Reply:
Otherwise, this does not relate to this PR, right  ?

Reply:
Sorry, I missed that PR- it does relate, because the `select VIDEO_MCUX_CSI` statement here will trigger a Kconfig error on the RT11xx series when building the CSI2X driver, because `VIDEO_MCUX_CSI` currently has a dependency on `HAS_MCUX_CSI`. All this means is that this PR is dependent on #71860, as far as I can tell

Reply:
But this CSI2Rx driver doesn't build. Both CSI and CSI2Rx will build only when there are sensor connected to them and it is triggered in the sensor overlay. By default, CSI2Rx and CSI nodes are disabled (to save power).

So it is ok to approve the PR alone or there's something I need to do ?

Reply:
@danieldegrasse : So, should we move the commit https://github.com/zephyrproject-rtos/zephyr/pull/72420/commits/b226dedfbc2c4cbcf37dd1c86e85895a4660afa3 here ?

Reply:
I'm fine to move b226dedfbc2c4cbcf37dd1c86e85895a4660afa3 into #71860.

> But this CSI2Rx driver doesn't build. Both CSI and CSI2Rx will build only when there is a camera sensor connected to them and the build is triggered in the camera sensor overlay (another PR). By default, CSI2Rx and CSI nodes are disabled.

Maybe I'm not following here- are you saying that since the CSI2Rx driver does not build as part of this PR, there is no dependency on #71860? Generally, we want to be able to build code that is introduced with a PR as part of that PR. Finding out there is a build error later on means we have to fix something in the tree, which is less clean than getting it right the first time around

Reply:
Okay ...

- Added build test for mipi csi2rx
- Split https://github.com/zephyrproject-rtos/zephyr/pull/72623 from https://github.com/zephyrproject-rtos/zephyr/pull/72420 to avoid mutual dependency

This PR is now dependent on https://github.com/zephyrproject-rtos/zephyr/pull/72623 and https://github.com/zephyrproject-rtos/zephyr/pull/71854.

Is it ok for you now ?

Reply:
Yes, I think this is ok. To make sure I understand the dependency chain here:

#71859: dependent on #72623 and #71854
#71854: no dependencies
#72420: dependent on #71859 (and its dependencies)
#72434: dependent on #72420 (and its dependencies), and #71854
#72632: dependent on #72434 (and its dependencies)

Is this all accurate?

Reply:
@danieldegrasse It's nearly correct. The dependency chain are:
https://github.com/zephyrproject-rtos/zephyr/pull/72435: no dependencies
https://github.com/zephyrproject-rtos/zephyr/pull/71854: no dependencies
https://github.com/zephyrproject-rtos/zephyr/pull/72623: no dependencies
https://github.com/zephyrproject-rtos/zephyr/pull/71859: dependent on https://github.com/zephyrproject-rtos/zephyr/pull/72623 and https://github.com/zephyrproject-rtos/zephyr/pull/71854
https://github.com/zephyrproject-rtos/zephyr/pull/72420: dependent on https://github.com/zephyrproject-rtos/zephyr/pull/71859 (and its dependencies).
https://github.com/zephyrproject-rtos/zephyr/pull/72434: dependent on https://github.com/zephyrproject-rtos/zephyr/pull/71854 and https://github.com/zephyrproject-rtos/zephyr/pull/71859 (and its dependency)
https://github.com/zephyrproject-rtos/zephyr/pull/72632: dependent on https://github.com/zephyrproject-rtos/zephyr/pull/72434 (and its dependencies)
https://github.com/zephyrproject-rtos/zephyr/pull/72633: dependent on https://github.com/zephyrproject-rtos/zephyr/pull/71859 (and its dependencies) and https://github.com/zephyrproject-rtos/zephyr/pull/72434 (and its dependency)

Reply:
Forgot : https://github.com/zephyrproject-rtos/zephyr/pull/72631: no dependencies